.github/workflows/upgrade-deps.yml

```diff
@@ -0,0 +1,60 @@
+name: Upgrade dependencies
+
+on:
+  workflow_dispatch: # Allow running on-demand
+  schedule:
+    # Every Monday at 8:00 UTC (4:00 Eastern)
+    - cron: '0 8 * * 1'
+
+jobs:
+  upgrade:
+    name: Upgrade & Open Pull Request
+    runs-on: ubuntu-latest
+    env:
+      BRANCH_NAME: auto-dependency-upgrades
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Use a separate key so the resulting PR also runs GH Actions
+          # https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
+      - uses: ./.github/setup-docker-compose
+      - name: Upgrade JS dependencies
+        run: >
+          docker-compose run --no-deps web /bin/sh -c 'rm -f yarn.lock && chown
+          -R $(whoami):$(whoami) . && npx --yes yarn-upgrade-all'
+      - name: Upgrade Python dependencies
+        run: >
+          docker-compose run --no-deps web /bin/sh -c 'pip-compile --upgrade -o
+          requirements/prod.txt requirements/prod.in && pip-compile --upgrade -o
+          requirements/dev.txt requirements/dev.in'
+      - name: Lint
+        run: docker-compose run --no-deps web yarn lint
+      - name: Detect changes
+        id: changes
+        run:
+          echo "::set-output name=count::$(git status --porcelain=v1 2>/dev/null
+          | wc -l)"
+      - name: Commit & push changes
+        if: steps.changes.outputs.count > 0
+        run: |
+          sudo chown -R $(whoami):$(whoami) .git
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git commit -m "Automated dependency upgrades"
+          git push -f origin ${{ github.ref_name }}:$BRANCH_NAME
```

Comment:
@jerivas It looks to me like this process isn't doing what we intended. Linting passed in https://github.com/oddbird/Metecho/runs/6213681216 but then failed in the new PR: https://github.com/oddbird/Metecho/pull/2148. I'm wondering if this is running with the original dependencies installed, not the upgraded dependencies?

It's not really a big deal, so we could just remove these three steps from the workflow.

Reply:
The detect & commit steps are there for requirements/package.json so we want to keep those. It does seem like the `lint` step is running with old deps for some reason. I'll take a look because it would also impact the other projects that use this workflow.

Reply:
Makes sense. I think the only purpose of the lint step in this context (since it will run again on the newly-created PR) was to commit formatting changes if e.g. Prettier or Black is upgraded. So it's kinda nice in theory, but also somewhat minor as a use-case.